Robot2018/src/org/usfirst/frc/team199/Robot2018/RobotMap.java

```diff
@@ -178,4 +178,15 @@ public int getPort(String key, int def) {
 		return (int) SmartDashboard.getNumber("Port/" + key, def);
 	}
 
+	/**
+	 * Uses SmartDashboard and math to calculate a *great* default kD
+	 */
+	public double calcDefkD() {
+		double timeConstant = Robot.getConst("Omega Max", 5330) * Robot.getConst("Mass of Robot", 54.4311)
+				* Robot.getConst("Radius of Drivetrain Wheel", 0.0635)
+				* Robot.getConst("Radius of Drivetrain Wheel", 0.0635) / Robot.getConst("Stall Torque", 2.41);
+		double cycleTime = Robot.getConst("Code cycle time", 0.1);
```

Comment:
Where did this default come from? Check the default period for the PIDController.